rsts/deployment/cluster_config/performance.rst

```diff
@@ -13,16 +13,49 @@ The video below contains an overview of the Flyte architecture, what is meant by
 Scaling up FlytePropeller
 ==========================
 `FlytePropeller <https://pkg.go.dev/github.com/flyteorg/flytepropeller>`_ is the core engine of Flyte that executes the workflows for Flyte. It is implemented as a `controller <https://kubernetes.io/docs/concepts/architecture/controller/>`__ in Kubernetes.
+It is designed as a `Kubernetes Controller <https://kubernetes.io/docs/concepts/architecture/controller/>`_, where the desired state is specified as a FlyteWorkflow `Custom Resource <https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/>`_.
+
+One of the base assumptions of FlytePropeller is that every workflow is independent and can be executed by a completely distinct process, without a need for communication with other processes. On the flip-side, one workflow is tracks the dependencies using a DAG structure and hence constantly needs synchronization.
```

Comment:
is tracks -> is tracking 
Also which worklfow is this that tracks the dependencies.